DOC-5711 - current docs cleanup

jamesnocentini opened the pull request:
https://issues.couchbase.com/browse/DOC-5711

Changes:

- move import processing paragraph to `import_docs` prop (from `enable_shared_bucket_access`)
- break first paragraph in `import_docs` intro
- review paragraphs under `import_docs` (Tombstones and Implementation notes for XATTRs)
- fix up external links in the `enable_shared_bucket_access` description

jamesnocentini committed:
DOC-5711 - current docs cleanup

https://issues.couchbase.com/browse/DOC-5711

Changes:

- move import processing paragraph to `import_docs` prop (from `enable_shared_bucket_access`)
- break first paragraph in `import_docs` intro
- review paragraphs under `import_docs` (Tombstones and Implementation notes for XATTRs)
- fix up external links in the `enable_shared_bucket_access` description

jamiltz commented:
@rajagp For your information, this PR is a small clean up of current docs. Please let me know if you have any feedback on this. Many thanks.

rajagp commented on modules/ROOT/assets/attachments/sg.yaml:
```diff
@@ -142,9 +142,33 @@ properties:
 
               `enable_shared_bucket_access` needs to be `true` on all nodes participating in such a configuration where as `import_docs` can only be set to `true` on a single node.
+
+              Starting Sync Gateway with `import_docs` enabled will generate the mobile-specific metadata for all the pre-existing documents in the Couchbase Server bucket. From then on, documents can be inserted on the Server directly (with N1QL or SDKs) or through the Sync Gateway REST API.
```
```suggestion
              Starting Sync Gateway with `import_docs` enabled will generate the mobile-specific metadata for all the existing documents in the Couchbase Server bucket. With shared_bucked_access enabled, documents can be mutated on the Server directly (with N1QL or SDKs) or through the Sync Gateway REST API and these changes will be seamlessly synchronized over to the connected Couchbase Lite enabled clients.
```

rajagp commented on modules/ROOT/assets/attachments/sg.yaml:
```diff
@@ -142,9 +142,33 @@ properties:
+
+              The [Import+](#1.5/log) log key can be used to troubleshoot the import process further in the logs.
+
+              A filter function can be specified using the [databases.foo_db.import_filter](#1.5/databases-foo_db-import_filter) property to only import specific documents.
```
The entire "Import Process" section should be moved to https://docs.couchbase.com/sync-gateway/2.6/shared-bucket-access.html. Its a concept that we don't want buried in config specs. Also, please rename "Mobile-Web Data Sync" navigation menu to "Mobile-Server Data Sync"

ibsoln replied:
Done in PR#140 https://github.com/couchbase/docs-sync-gateway/pull/140/files/0feceacb42434bab2415d837210010ca0f4f8fb4

rajagp commented on modules/ROOT/assets/attachments/sg.yaml:
```diff
@@ -142,9 +142,33 @@ properties:
+
+              A filter function can be specified using the [databases.foo_db.import_filter](#1.5/databases-foo_db-import_filter) property to only import specific documents.
+
+              ##### Import and Sync Gateway Accelerator
```
Remove this section. SG Accelerator is REMOVED and not a product offering. Please remove all references in docs

rajagp commented on modules/ROOT/assets/attachments/sg.yaml:
```diff
@@ -142,9 +142,33 @@ properties:
+
+              ##### Import and Sync Gateway Accelerator
+
+              In a deployment with Sync Gateway Accelerator, feed-based import processing is performed by the Accelerator nodes. Import work is sharded across the Accelerator cluster, in the same way that indexing work is sharded today.
```
Remove all references to Accelerator. Its not a product offering

rajagp commented on modules/ROOT/assets/attachments/sg.yaml:
```diff
@@ -304,43 +328,26 @@ properties:
-              Import is the name for the process by which non-Sync Gateway writes obtain mobile metadata. Any non-Sync Gateway write is eligible for import. A filter function can be specified using the [databases.foo_db.import_filter](#1.5/databases-foo_db-import_filter) property to only import specific documents.
-
-              The document is first run through the Sync Function to compute read security and routing, with the following specifics/differences:
+              Introduced in Sync Gateway 1.5, this property specifies whether to enable Mobile-Web Data Sync.
```
```suggestion
              Introduced in Sync Gateway 1.5, this property specifies whether to enable Mobile-Server Data Sync a.k.a "mobile convergence". You can learn more about this functionality [here](https://docs.couchbase.com/sync-gateway/current/shared-bucket-access.html)
```

rajagp commented on modules/ROOT/assets/attachments/sg.yaml:
```diff
@@ -304,43 +328,26 @@ properties:
-              When convergence is enabled, mobile tombstones are now also server tombstones. The document body is deleted, and the information required to replicate the tombstone is retained in the mobile extended attribute.
-
-              ##### Metadata Purge Interval
+              When `enable_shared_bucket_access` is enabled, mobile tombstones are now also server tombstones. The document body is deleted, and the information required to replicate the tombstone is retained in the mobile extended attribute.
```
```suggestion
              When `enable_shared_bucket_access` is enabled, mobile tombstones are now also server tombstones. The document body is deleted, and only the mobile sync metadata required to replicate the tombstone is retained in the mobile extended attribute.
```

rajagp replied:
Move the "tombstone" section over to https://docs.couchbase.com/sync-gateway/2.6/shared-bucket-access.html. Its a core concept

rajagp commented on modules/ROOT/assets/attachments/sg.yaml:
```diff
@@ -304,43 +328,26 @@ properties:
-              ##### Metadata Purge Interval
+              When `enable_shared_bucket_access` is enabled, mobile tombstones are now also server tombstones. The document body is deleted, and the information required to replicate the tombstone is retained in the mobile extended attribute.
 
               The server's metadata purge interval becomes an important consideration for mobile deployments under convergence. When the server purges a tombstone (based on the metadata purge interval), that tombstone will no longer be replicated to mobile clients.
```
Move the Metadata Purge Interval Section to https://docs.couchbase.com/sync-gateway/2.6/shared-bucket-access.html

rajagp commented on modules/ROOT/assets/attachments/sg.yaml:
```diff
@@ -304,43 +328,26 @@ properties:
               The server's metadata purge interval becomes an important consideration for mobile deployments under convergence. When the server purges a tombstone (based on the metadata purge interval), that tombstone will no longer be replicated to mobile clients.
 
-              For most mobile deployments, customers should set the server's metadata purge interval based on their expected client replication frequency, to ensure that clients are notified of the tombstone prior to that tombstone being purged.
+              For most mobile deployments, users should set the server's metadata purge interval based on their expected client replication frequency, to ensure that clients are notified of the tombstone prior to that tombstone being purged.
```
```suggestion
              Users should set the server's metadata purge interval based on their expected client partition window, to ensure that clients are notified of the tombstone prior to that tombstone being purged on the server.
```

rajagp reviewed (dismissed):
@jamiltz  Please note that some of the content needs to be moved out to a different page. This is important as these sections are impacted by the "write scale enhancements" that are in place for 2.7

Commit:
Apply suggestions from code review

Co-Authored-By: Priya Rajagopal <rajagp@users.noreply.github.com>

ibsoln committed:
Merge branch 'release/2.6' into doc-5711-current-docs-cleanup

jamesnocentini committed:
Doc 5711 current docs cleanup (#141)

* https://issues.couchbase.com/browse/DOC-5354

- fix link in `allow_conflicts` description

* https://issues.couchbase.com/browse/DOC-5354

- fix link in `delta_sync` description

* https://issues.couchbase.com/browse/DOC-5354

- fix doc links

* https://issues.couchbase.com/browse/DOC-5354

- fix broken link in `default` description

* DOC-5737 - Access Control edits (#107)

https://issues.couchbase.com/browse/DOC-5737

Information architecture:

- move info about network ports from Authorization page to Network Ports section on Getting Started page
- move concept intros from Sync Function API page to Authorization page
- move "Sync Function API" page to Access Control section
- add Anonymous Access section in authentication.adoc
- add remaining content from authorizing-users.adoc to write-security.adoc
- remove authorizing-users.adoc
- create Users and Roles page

Content editing:

- rename 'Introduction to Channels' title to 'Introduction'
- rename 'Replicating channels to Couchbase Lite' title to 'Replicating Channels' and make it h2
- rename 'Removing a replicated document' title to 'Access Removal' and make it h2
- add intro paragraph on Sync Function API page
- re-word Add Access > Sync Function intro paragraph
- re-word paragraph in Add to Channel
- add paragraph in Data Validation section
- re-word Write Access intro paragraph
- move content about user registration to Users and Roles
- mention oidc and custom auth

Nav and links:

- update nav
- add page alias
- rename sync-function-api to sync-function
- rename data-routing to sync-gateway-channels
- rename write-access to write-security
- update xrefs
- update links in config reference

* https://issues.couchbase.com/browse/DOC-5785

* Update supported-os.adoc (#116)

Extended the deprecation period from 2 releases to 4 releases to give sufficient time for customers to migrate with our new release schedule.

* DOC-5889 - overview of sync gateway (#113)

https://issues.couchbase.com/browse/DOC-5889

Changes:

- Add diagram
- Add text
- Group server nodes in a rectangle (cluster) per feedback
- Lower number of arrows per feedback
- Add "port" suffix to make it clear it's the port number per feedback

* bump :xref-cb-config: to 2.6

* DOC-5893 - sg access paths (#121)

https://issues.couchbase.com/browse/DOC-5893

As part of DOC-5893, updating the diagram for the Mobile-Web Data Sync page

Check-in source file (cbm-architecture.sketch)

* DOC-5789-IPv6-Server-Connection-Examples (#124)

https://issues.couchbase.com/browse/DOC-5789

* DOC-6038-Config-Changes (#129)

https://issues.couchbase.com/browse/DOC-6038

* DOC-6036 - attributes file (#126)

https://issues.couchbase.com/browse/DOC-6036 

Changes:

- Add _attributes.adoc file
- Add :version: attribute
- Add :version-full: attribute
- Use :version-full: in getting-started.adoc
- Use :version-full: in stats-monitoring.adoc

ibsoln committed:
DOC-5711-Current-Docs-Cleanup https://github.com/couchbase/docs-sync-gateway/pull/111 (#140)

Addresses comments raised in response to above pull request

ibsoln reviewed (approved):
Changed as requested
